bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.i18n.UnitProvider;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.ImperialUnits;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.weathercompany.internal.config.WeatherCompanyConfig;
+import org.openhab.binding.weathercompany.internal.model.DayPart;
+import org.openhab.binding.weathercompany.internal.model.Forecast;
+import org.openhab.binding.weathercompany.internal.model.PwsObservations;
+import org.openhab.binding.weathercompany.internal.model.PwsObservations.Observations;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link WeatherCompanyHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * API documentation is located here
+ * - https://docs.google.com/document/d/1eKCnKXI9xnoMGRRzOL1xPCBihNV2rOet08qpE_gArAY/edit
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class WeatherCompanyHandler extends BaseThingHandler {
+    // Five-day weather forecast URL
+    private static final String BASE_FORECAST_URL = "https://api.weather.com/v3/wx/forecast/daily/5day";
+
+    // Personal Weather Station observations URL
+    private static final String BASE_PWS_URL = "https://api.weather.com/v2/pws/observations/current";
+
+    public static final int WEATHER_COMPANY_API_TIMEOUT = 10000;
+    public static final int REFRESH_JOB_INITIAL_DELAY = 3;
+
+    // Used to sanitize the API key in the URL in debug log messages
+    private static final String REPLACE_API_KEY = "XXXXXXXXXXXXXXXXXXXXX";
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherCompanyHandler.class);
+
+    private final Gson gson = new GsonBuilder().serializeNulls().create();
+
+    // Provided by handler factory
+    private @NonNullByDefault({}) TimeZoneProvider timeZoneProvider;
```

Comment:
Given they compare in through the constructor I would make them final

And please don't use this annotation hack except for References

Reply:
> Given they compare in through the constructor I would make them final

Good catch.

> And please don't use this annotation hack except for References

Ok. I had to rework a bit of the code to resolve another usage of this.